maintainers_annotation/maintainers_annotation.go

```diff
@@ -0,0 +1,190 @@
+	// that is not supported by Scorecard. E.g. Clang-Tidy is used as SAST tool but not identified
+	// because its not supported.
+	AnnotationNotSupported AnnotationReason = "not-supported"
+	// AnnotationNotDetected is to annotate when the maintainer fulfills a check or probe in a way
+	// that is supported by Scorecard but not identified. E.g. Dependabot is configured in the
+	// repository settings and not in a file.
+	AnnotationNotDetected AnnotationReason = "not-detected"
+)
+
+// AnnotationReasonDoc maps the reasons to their human-readable explanations.
+var AnnotationReasonDoc = map[AnnotationReason]string{
+	AnnotationTestData:      "The files or code snippets are only used for test or example purposes.",
+	AnnotationRemediated:    "The dangerous files or code snippets are necessary but remediations were already applied.",
+	AnnotationNotApplicable: "The check or probe is not applicable in this case.",
+	AnnotationNotSupported:  "The check or probe is fulfilled but in a way that is not supported by Scorecard.",
+	AnnotationNotDetected:   "The check or probe is fulfilled but in a way that is supported by Scorecard but it was not detected.",
+}
+
+// Annotation groups the annotation reason and, in the future, the related probe.
+// If there is a probe, the reason applies to the probe.
+// If there is not a probe, the reason applies to the check or group of checks in
+// the exemption.
+type Annotation struct {
+	AnnotationReason AnnotationReason `yaml:"annotation"`
+}
+
+// Exemption defines a group of checks that are being annotated for various reasons.
+type Exemption struct {
+	Checks      []string     `yaml:"checks"`
+	Annotations []Annotation `yaml:"annotations"`
+}
+
+// MaintainersAnnotation is a group of exemptions.
+type MaintainersAnnotation struct {
+	Exemptions []Exemption `yaml:"exemptions"`
+}
+
+// parseScorecardYmlFile takes the scorecard.yml file content and returns a `MaintainersAnnotation`.
+func parseScorecardYmlFile(ma *MaintainersAnnotation, content []byte) error {
+	unmarshalErr := yaml.Unmarshal(content, ma)
+	if unmarshalErr != nil {
+		return sce.WithMessage(sce.ErrScorecardInternal, unmarshalErr.Error())
+	}
+
+	return nil
+}
+
+// validateAndParseScorecardYmlFile takes the scorecard.yml files found, validates if its the correct
+// configuration file and parse its configurations.
+func validateAndParseScorecardYmlFile(path string, content []byte, args ...interface{}) (bool, error) {
+	// Verify if received the necessary parameters to save the scorecard YML configurations
+	if len(args) != 1 {
+		return false, fmt.Errorf(
+			"validateAndParseScorecardYmlFile requires exactly one argument to save maintainers annotation data")
+	}
+	ma, ok := args[0].(*MaintainersAnnotation)
+	if !ok {
+		return false, fmt.Errorf(
+			"validateAndParseScorecardYmlFile requires argument of type *MaintainersAnnotation")
+	}
+
+	// Exclude scorecard GitHub Action file
+	if strings.Contains(path, ".github") {
```

Comment:
```suggestion
	if strings.Contains(path, ".github/workflows") {
```

In case we wish to support saving the .scorecard.yml file in .github/.

Reply:
My understand is that we wanted to support:

File location:
If it's a repository, in the repository root directory
If it's a GitHub organization, in the root directory of the .github repository for the organization

Not sure if this change satisfies the second option or in a normal repository to put the scorecard.yml file in .github/ folder.

Reply:
Any action items here?